nixos/tests/lxd.nix

```diff
@@ -6,16 +6,47 @@ let
   #
   # I've chosen to import Alpine Linux, because its image is turbo-tiny and,
   # generally, sufficient for our tests.
-  alpine-meta = pkgs.fetchurl {
+  alpine-meta-x86 = pkgs.fetchurl {
     url = "https://tarballs.nixos.org/alpine/3.12/lxd.tar.xz";
     hash = "sha256-1tcKaO9lOkvqfmG/7FMbfAEToAuFy2YMewS8ysBKuLA=";
   };
+  alpine-meta-for = arch: pkgs.stdenv.mkDerivation {
+    name = "alpine-meta-${arch}";
+    version = "3.12";
+    unpackPhase = "true";
+    buildPhase = ''
+      runHook preBuild
 
-  alpine-rootfs = pkgs.fetchurl {
-    url = "https://tarballs.nixos.org/alpine/3.12/rootfs.tar.xz";
-    hash = "sha256-Tba9sSoaiMtQLY45u7p5DMqXTSDgs/763L/SQp0bkCA=";
+      tar xvf ${alpine-meta-x86}
+      sed -i 's/architecture: .*/architecture: ${arch}/' metadata.yaml
+
+      runHook postBuild
+    '';
+    installPhase = ''
+      runHook preInstall
+
+      tar czRf $out *
+
+      runHook postInstall
+    '';
   };
 
+  alpine-meta = {
+    x86_64-linux = alpine-meta-x86;
+    aarch64-linux = alpine-meta-for "aarch64";
+  }.${pkgs.system} or (throw "Unsupported system: ${pkgs.system}");
+
+  alpine-rootfs = {
+    x86_64-linux = pkgs.fetchurl {
+      url = "https://tarballs.nixos.org/alpine/3.12/rootfs.tar.xz";
+      hash = "sha256-Tba9sSoaiMtQLY45u7p5DMqXTSDgs/763L/SQp0bkCA=";
+    };
+    aarch64-linux = pkgs.fetchurl {
+      url = "https://dl-cdn.alpinelinux.org/alpine/v3.15/releases/aarch64/alpine-minirootfs-3.15.4-aarch64.tar.gz";
```

Comment:
When I was introducing this test, [it was suggested](https://github.com/NixOS/nixpkgs/pull/97550#issuecomment-689653841) that we upload the image at tarballs.nixos.org instead, as to not rely on someone else's servers for our tests.

@AmineChikhaoui can you upload that image to tarballs.nixos.org? 🙂

Reply:
That works for me

If you're otherwise ok with this change, perhaps we should merge as-is and update it once someone has uploaded the image to tarballs.nixos.org?